src/test/java/com/kainos/ea/service/JobSpecificationsTest.java

```diff
@@ -0,0 +1,55 @@
+package com.kainos.ea.service;
+
+
+import com.kainos.ea.dao.RolesDao;
+import com.kainos.ea.database.DatabaseConnection;
+import com.kainos.ea.exception.DatabaseConnectionException;
+import com.kainos.ea.exception.RoleNotExistException;
+import com.kainos.ea.model.JobSpecification;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class JobSpecificationsTest {
+
+    RolesDao rolesDao = Mockito.mock(RolesDao.class);
+
+    DatabaseConnection databaseConnector = Mockito.mock(DatabaseConnection.class);
+
+    RolesService rolesService = new RolesService(rolesDao, databaseConnector);
+    Connection conn;
+
+    @Test
+    void getAllRoles_shouldReturnListOfSpecifications_whenDaoReturnsListOfSpecifications() throws SQLException, DatabaseConnectionException, IOException, RoleNotExistException {
```

Comment:
these tests should be moved to the roles service class

Reply:
done